File: app/src/main/java/vng/zalo/tdtai/zalo/zalo/models/Message.java
package vng.zalo.tdtai.zalo.zalo.models;

import com.google.firebase.Timestamp;
import java.util.HashMap;
import java.util.Map;

import vng.zalo.tdtai.zalo.zalo.utils.Constants;

public class Message extends DataModel {
    public String content;
    public Timestamp createdTime;
    public String senderPhone;
    public String senderAvatar;
    public int type;

    public Message(){}

    public final Map<String, Object> toMap(){
        Map<String, Object> res = new HashMap<>();

        res.put("content",content);
        res.put("createdTime",createdTime);
        res.put("senderPhone",senderPhone);
        res.put("type", type);

        return res;
    }
}

File: app/src/main/java/vng/zalo/tdtai/zalo/zalo/models/Message.kt
package vng.zalo.tdtai.zalo.zalo.models

import com.google.firebase.Timestamp
import java.util.*

data class Message(var content: String? = null,
                   var createdTime: Timestamp? = null,
                   var senderPhone: String? = null,
                   var senderAvatar: String?=null,
                   var type: Int? = null) {
    fun toMap(): Map<String, Any?> {
        return HashMap<String, Any?>().apply {
            put("content", content)
            put("createdTime", createdTime)
            put("senderPhone", senderPhone)
            put("type", type)
        }
    }
}